### PR TITLE
🐛 cron/workspace-cleaner: only run task for docker environments

### PR DESCRIPTION
**Pull request by pedroslopez**
## What

Running cron on a kubernetes environment was causing an error when executing the WorkspaceCleaner:

```
2022-09-23 19:39:59 ERROR i.m.s.DefaultTaskExceptionHandler(handle):47 - Error invoking scheduled task for bean [io.airbyte.cron.selfhealing.WorkspaceCleaner@774358c5] /workspace
java.nio.file.NoSuchFileException: /workspace
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:92) ~[?:?]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106) ~[?:?]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111) ~[?:?]
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55) ~[?:?]
	at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:148) ~[?:?]
	at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99) ~[?:?]
	at java.nio.file.Files.readAttributes(Files.java:1849) ~[?:?]
	at java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:220) ~[?:?]
	at java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:277) ~[?:?]
	at java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:323) ~[?:?]
	at java.nio.file.FileTreeIterator.<init>(FileTreeIterator.java:71) ~[?:?]
	at java.nio.file.Files.walk(Files.java:3917) ~[?:?]
	at java.nio.file.Files.walk(Files.java:3972) ~[?:?]
	at io.airbyte.cron.selfhealing.WorkspaceCleaner.deleteOldFiles(WorkspaceCleaner.java:51) ~[io.airbyte-airbyte-cron-0.40.6.jar:?]
	at io.airbyte.cron.selfhealing.$WorkspaceCleaner$Definition$Exec.dispatch(Unknown Source) ~[io.airbyte-airbyte-cron-0.40.6.jar:?]
	at io.micronaut.context.AbstractExecutableMethodsDefinition$DispatchedExecutableMethod.invoke(AbstractExecutableMethodsDefinition.java:378) ~[micronaut-inject-3.6.3.jar:3.6.3]
	at io.micronaut.inject.DelegatingExecutableMethod.invoke(DelegatingExecutableMethod.java:76) ~[micronaut-inject-3.6.3.jar:3.6.3]
	at io.micronaut.scheduling.processor.ScheduledMethodProcessor.lambda$process$5(ScheduledMethodProcessor.java:127) ~[micronaut-context-3.6.3.jar:3.6.3]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577) ~[?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:358) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.lang.Thread.run(Thread.java:1589) ~[?:?]

```

## How

Since this is only intended to run on docker contexts, use the micronaut environment to determine if the task should run.

### CODE REVIEW

File: airbyte-cron/src/main/resources/application.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+micronaut:
+  application:
+    name: airbyte-cron
+
+airbyte:
+  worker:
+    env: ${WORKER_ENVIRONMENT:DOCKER}
```
Comment (pedroslopez) [inline airbyte-cron/src/main/resources/application.yml:7]: Using the same structure as in https://github.com/airbytehq/airbyte/blob/c4fb6b7f9eb9c371902e8c4bec484c2427954cf4/airbyte-workers/src/main/resources/application.yml#L87-L88

**Review by gosusnp — COMMENTED** [review]
The require part looks good to me. I am not sure about the config itself, we had some discussions about that.
@benmoriceau, what do you think?

**Review by benmoriceau — CHANGES_REQUESTED** [review]
LGTM but we need to add the new env variable to the file `kube/overlays/resources/cron.yml` .

And we will need to make sure that the helm chart have this value as well when they got created.

**Comment by pedroslopez** [comment]
@benmoriceau  Added to kube/resources/cron.yml in [bbe301b](https://github.com/airbytehq/airbyte/pull/17168/commits/bbe301b636f60a5c2ba841310dc10e8201cb6a65)

**Comment by pedroslopez** [comment]
Updated to use micronaut environments cc @jdpgrailsdev

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Review by benmoriceau — APPROVED** [review]

**Pull request merged**